Assets/LeapMotionModules/UI/Interaction/InteractionHand.cs

```diff
@@ -0,0 +1,780 @@
+﻿using Leap.Unity.RuntimeGizmos;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Leap.Unity.UI.Interaction {
+
+  public class InteractionHand {
+
+    public InteractionManager interactionManager;
+
+    private Func<Hand> _handAccessor;
+    private Hand _hand;
+    private bool _isLeft;
+
+    public InteractionHand(InteractionManager interactionManager,
+                           Func<Hand> handAccessor,
+                           Chirality handedness,
+                           float hoverActivationRadius,
+                           float touchActivationRadius) {
+      this.interactionManager = interactionManager;
+      _handAccessor = handAccessor;
+      _isLeft = handedness == Chirality.Left;
+
+      InitHovering(hoverActivationRadius);
+      InitTouch(touchActivationRadius);
+      InitContact();
+      InitGrasping();
+    }
+
+    public Hand GetLeapHand() {
+      return _hand;
+    }
+
+    public void FixedUpdateHand(bool doHovering, bool doContact, bool doGrasping) {
+      _hand = _handAccessor();
+
+      if (doHovering) FixedUpdateHovering();
```

Comment:
I would argue it makes sense to add more strict ordering to the ordering.  Currently it looks like callbacks for objects are dispatched in 'hand order'.  So you receive all possible callbacks from one hand, then all possible callbacks from the next hand.  This can result in strange cases where my grasped state is not the same between my first and second hover stay callback during the same frame!

I would propose ordering by concept instead.  So something like, first all hover callbacks, then all touch callbacks, then all grasp callbacks.